test/WaveOps/WaveActiveSum.fp16.test

```diff
@@ -0,0 +1,178 @@
+#--- source.hlsl
+StructuredBuffer<half4> In  : register(t0);
+RWStructuredBuffer<half4> Out1 : register(u1); // test scalar
+RWStructuredBuffer<half4> Out2 : register(u2); // test half2
+RWStructuredBuffer<half4> Out3 : register(u3); // test half3
+RWStructuredBuffer<half4> Out4 : register(u4); // test half4
+RWStructuredBuffer<half4> Out5 : register(u5); // constant folding
+
+[numthreads(4,1,1)]
+void main(uint3 tid : SV_GroupThreadID)
+{
+    half4 v = In[0];
+
+    // Mask per "active lane set": only <=N lanes contribute
+    half s1 = WaveActiveSum((tid.x <= 0) ? v.x : 0);
+    half s2 = WaveActiveSum((tid.x <= 1) ? v.x : 0);
+    half s3 = WaveActiveSum((tid.x <= 2) ? v.x : 0);
+    half s4 = WaveActiveSum((tid.x <= 3) ? v.x : 0);

```

Comment:
I think we probably should be putting the WaveActiveSum calls under `if` statements rather than using the ternary operator on the operands. Putting it under an `if` statement changes which threads are active and thus participating in the wave operation, whereas the ternary operator on the argument just changes the input value.